bin/circle/search-circle-builds

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env ruby
+# A script to analyze a range of CircleCI builds for known failures.
+# For available options, run ./search-circle-builds -h
+
+require 'pp'
+require 'json'
+require 'memoist'
+require 'open-uri'
+require 'optparse'
+
+CIRCLE_API_BASE = 'https://circleci.com/api/v1.1/project/github/code-dot-org/code-dot-org'
+
+KNOWN_FAIL_TYPES = [
+    {
+        search_string: 'ConnectionTimeoutError',
+        step_string: 'run_tests',
+        container: 1
+    },
+    {
+        search_string: 'io_fillbuf',
+        step_string: 'run_tests',
+        container: 1
+    },
+]
+
+extend Memoist
+
+def get_build(id)
+  JSON.parse(open("#{CIRCLE_API_BASE}/#{id}").read)
+end
+
+memoize :get_build
+
+def output_url(build, container, step_includes)
+  build['steps'].select { |o| o['name'].include? step_includes }[0]['actions'][container]['output_url']
+end
+
+memoize :output_url
+
+def get_output(id, container, step_includes)
+  JSON.parse(open(output_url(get_build(id), container, step_includes)).read)
+rescue => _
+end
```

Comment:
This might be a good place in the stack to document what all these parameters mean. I was dimly aware of the output format shown at https://circleci.com/api/v1.1/project/github/code-dot-org/code-dot-org/22255 and still had some trouble following this